src/common/select-response.ts

```diff
@@ -1,11 +1,41 @@
 import { defaultDelay, defaultStatus } from "../constants";
 import { type Mock, type MockResponse } from "../mockfile";
 
+/**
+ * Enforce all headers in object to be in lower case. This is typically not a problem in node mode,
+ * since Express do this, but is needed if you run the mocks in browser mode
+ * */
+function enforceLowerCaseHeaders(
+    headers: undefined | Record<string, string | string[] | undefined>,
+): Record<string, string | string[] | undefined> {
+    const lowercase = {};
+    if (headers) {
+        for (const [key, value] of Object.entries(headers)) {
+            lowercase[key.toLocaleLowerCase()] = value;
+        }
+    }
+    return lowercase;
+}
+
 /** Append response with default data if missing */
-function normalizeResponse(response: MockResponse): MockResponse {
-    response.status ??= defaultStatus;
-    response.delay ??= defaultDelay;
-    return response;
+function normalizeResponse(
+    request: {
+        headers: Record<string, string | string[] | undefined>;
+        cookies: Record<string, string>;
+        body: unknown;
+    },
+    response: MockResponse,
+): MockResponse {
+    return {
+        status: defaultStatus,
+        delay: defaultDelay,
+        ...response,
+        body:
+            typeof response.body === "function"
```

Comment:
To slightly satisfy typescript and eslint:

```ts
type BodyFunction = (request: ...): string;

function isBodyFunction(value: unknown): value is BodyFunction {
  return typeof value === 'function';
}

return {
  body: isBodyFunction(response.body) ? response.body(request) : response.body;
}
```

It's not really safer if one is trying to lie to the type system as we still just rely on `typeof ... === 'function'` but I think that part is good enough.